### PR TITLE
Foundation deployment

### PR DESCRIPTION
**Pull request by microbit-matt-hillsdon**
I've deleted the upstream workflows rather than disable them for clarity. Will always be easy to
reinstate if needed.

Non-trivial aspects:
- The review/branch/PR previews model we use relies on a working base URL. There a few barriers to this:
    - A bespoke router redirects you to the root of the review domain so breaking refresh. I've added support for a base URL.
    - Absolute image paths. You can import images and have the bundler deal with them properly. This might be a bit disruptive though so perhaps we could do upstream 
        - I've done it here for now. Quite a few images are in new UX.
    - Font paths. They're relative in the `all.min.css` source so not entirely clear what's going wrong.
        - They work by fluke upstream. In current use `all.min.css` is deployed to the root but references the fonts with ../webfonts/... this works because ../ at the root is ignored but at any other path it breaks.  Maybe it was always intended to have a CSS directory? I've done that for now.
- There's lots in public/ that will need to have suboptimal cache settings as the bundler isn't aware so output file names don't include a hash. Perhaps one to follow up on before we go live.
- I've moved the 3D model from public/assets/model to just model as it wasn't managed by the bundler so can't get the same cache settings as other files in assets. `public/assets` is probably just not a good idea.

### CODE REVIEW

File: bin/print-ci-env.cjs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+process.env.STAGE = stage;
+// STAGE must be defined before this is imported
+const { bucketName, bucketPrefix } = require("../deployment.cjs");
+const baseUrl = !bucketPrefix ? "/" : `/${bucketPrefix}/`;
+const fullUrl = `https://${bucketName}${baseUrl}`;
+
+console.log(`STAGE=${stage}`);
+console.log(`VITE_STAGE=${stage}`);
+console.log(`BASE_URL=${baseUrl}`);
+console.log(`E2E_BASE_URL=${baseUrl}`);
```
Comment (microbit-matt-hillsdon) [inline bin/print-ci-env.cjs:22]: This is not needed here.

```suggestion
```

**Comment by microbit-matt-hillsdon** [comment]
The latest build is the first that isn't obviously broken: https://review-ml.microbit.org/deployment/

Will properly test the flows tomorrow.

File: src/imgs/stylised-battery-pack.svg
[the remark this answers is not in the document]
Comment (microbit-matt-hillsdon) [reply src/imgs/stylised-battery-pack.svg:1]: Just copied a few rather than moving. Fixed in 44c770b6edf449de30768fa625cdacba07544ca7

**Review by micque01 — APPROVED** [review]
LGTM, tested and it works nicely

**Pull request merged**